### ISSUE TITLE
Histogram for rank of logits not working when S >= 10.

### ISSUE THREAD

**Issue by esvhd** [esvhd/pypbo#1]
Error log. Need to look into `numpy` code. 

`numpy` version 0.19.1
`matplotlib` version 1.5.3

```python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-27-27a22a361a2f> in <module>()
      2 pbox = pbo.pbo(rtns_df, S=S, 
      3                metric_func=metric, threshold=1, n_jobs=4,
----> 4                plot=True, verbose=False, hist=True)

/Users/abc/git/pypbo/pypbo/pbo.py in pbo(M, S, metric_func, threshold, n_jobs, verbose, plot, hist)
    254                  R_rank, R_bar_rank,
    255                  rn, rn_bar,
--> 256                  w_bar,
    257                  logits,
    258                  R_n_star,

/Users/abc/git/pypbo/pypbo/pbo.py in plot_pbo(pbo_result, hist)
    338     sns.distplot(pbo_result.logits, rug=True, bins=10,
    339                  ax=axarr[1],
--> 340                  rug_kws={'color': 'r', 'alpha': .5},
    341                  kde_kws={'color': 'k', 'lw': 2., 'label': 'KDE'},
    342                  hist=hist,

/Users/abc/anaconda/envs/py35/lib/python3.5/site-packages/seaborn/distributions.py in distplot(a, bins, hist, kde, rug, fit, hist_kws, kde_kws, rug_kws, fit_kws, color, vertical, norm_hist, axlabel, label, ax)
    213         hist_color = hist_kws.pop("color", color)
    214         ax.hist(a, bins, orientation=orientation,
--> 215                 color=hist_color, **hist_kws)
    216         if hist_color != color:
    217             hist_kws["color"] = hist_color

/Users/abc/anaconda/envs/py35/lib/python3.5/site-packages/matplotlib/__init__.py in inner(ax, *args, **kwargs)
   1810                     warnings.warn(msg % (label_namer, func.__name__),
   1811                                   RuntimeWarning, stacklevel=2)
-> 1812             return func(ax, *args, **kwargs)
   1813         pre_doc = inner.__doc__
   1814         if pre_doc is None:

/Users/abc/anaconda/envs/py35/lib/python3.5/site-packages/matplotlib/axes/_axes.py in hist(self, x, bins, range, normed, weights, cumulative, bottom, histtype, align, orientation, rwidth, log, color, label, stacked, **kwargs)
   6008             # this will automatically overwrite bins,
   6009             # so that each histogram uses the same bins
-> 6010             m, bins = np.histogram(x[i], bins, weights=w[i], **hist_kwargs)
   6011             m = m.astype(float)  # causes problems later if it's an int
   6012             if mlast is None:

/Users/abc/anaconda/envs/py35/lib/python3.5/site-packages/numpy/lib/function_base.py in histogram(a, bins, range, normed, weights, density)
    503     if not np.all(np.isfinite([mn, mx])):
    504         raise ValueError(
--> 505             'range parameter must be finite.')
    506     if mn == mx:
    507         mn -= 0.5

ValueError: range parameter must be finite.
```

**Comment by nocluebutalotofit** [comment]
The problem is that the logits which get passed to distplot() tend to contain inf for S larger than 10.

The deeper question of course is where these infs come from and if they are a symptom of a larger issue but just to ignore them to get the distplot showing you could do something like this:

    # TODO hist is turned off at the moment. Error occurs when S is set to
    # a relatively large number, such as 16.
    pbo_result_logits_pd = pd.Series(pbo_result.logits).replace([np.inf, -np.inf], np.nan)
    sns.distplot(
        pbo_result_logits_pd.dropna(),
        rug=True,
        bins=10,
        ax=axarr[1],
        rug_kws={"color": "r", "alpha": 0.5},
        kde_kws={"color": "k", "lw": 2.0, "label": "KDE"},
        hist=hist,
        hist_kws={
            "histtype": "step",
            "linewidth": 2.0,
            "alpha": 0.7,
            "color": "g",
        },
    )

**Comment by nocluebutalotofit** [comment]
With this fix there is no need to remove the infs anymore.

**Comment by esvhd** [comment]
Thanks for looking into this, let me take a look as well.

**Comment by nocluebutalotofit** [comment]
This issue should be closed, since a59bbb8 fixed it, right?

**Comment by esvhd** [comment]
yep, closing
